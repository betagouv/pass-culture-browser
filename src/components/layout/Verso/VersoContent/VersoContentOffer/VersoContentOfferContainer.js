import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import { compose } from 'redux'
import { requestData } from 'redux-thunk-data'

import {
  selectBookables,
  selectBookingByRouterMatch
} from '../../../../../selectors/data/bookingsSelectors'
import { selectOfferByRouterMatch } from '../../../../../selectors/data/offersSelectors'
import { selectDistanceByRouterMatch } from '../../../../../selectors/geolocationSelectors'
import { selectIsNotBookableByRouterMatch } from '../../../../../selectors/isNotBookableSelector'
import { selectUserGeolocation } from '../../../../../selectors/geolocationSelectors'
import getStyle from './utils/getStyle'
import VersoContentOffer from './VersoContentOffer'

export const mapStateToProps = (state, ownProps) => {
  const { match } = ownProps
  const offer = selectOfferByRouterMatch(state, match) || {}
  const bookables = selectBookables(state, offer)
  const isNotBookable = selectIsNotBookableByRouterMatch(state, match)
  const { extraData } = offer || {}
  const style = getStyle(state, extraData)
  const booking = selectBookingByRouterMatch(state, match)
  const distance = selectDistanceByRouterMatch(state, match)
  const userGeolocation = selectUserGeolocation(state)
  const isCancelled = booking && booking.isCancelled

  return {
    bookables,
    booking,
    distance,
    isCancelled,
    isNotBookable,
    offer,
    style,
    userGeolocation
  }
}

const mapDispatchToProps = (dispatch, ownProps) => ({
  handleRequestMusicAndShowTypes: () => {
    const { musicTypes, showTypes } = ownProps

    if (!musicTypes) {
      dispatch(
        requestData({
          apiPath: '/musicTypes'
        })
      )
    }

    if (!showTypes) {
      dispatch(
        requestData({
          apiPath: '/showTypes'
        })
      )
    }
  }
})

export default compose(
  withRouter,
  connect(
    mapStateToProps,
    mapDispatchToProps
  )
)(VersoContentOffer)
