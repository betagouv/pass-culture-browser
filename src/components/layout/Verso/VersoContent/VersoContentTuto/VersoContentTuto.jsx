import PropTypes from 'prop-types'
import React from 'react'

const VersoContentTuto = ({ imageURL }) => (
  <div className="verso-info-tuto">
    <img
      alt="Et voilà ! Glissez la carte vers le bas pour revenir vers l’écran précédent, ou utilisez le bouton fermer. Bonne découverte !"
      className="verso-tuto-mediation is-full-width"
      src={imageURL}
    />
    <br />
    <br />
    <br />
    <br />
    <br />
    <br />
    <br />
    <br />
    <br />
    <br />
    <br />
  </div>
)

VersoContentTuto.propTypes = {
  imageURL: PropTypes.string.isRequired,
}

export default VersoContentTuto
