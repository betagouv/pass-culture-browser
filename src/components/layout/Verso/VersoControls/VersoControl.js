import React from 'react'
import PropTypes from 'prop-types'

import Finishable from '../../layout/Finishable'
import ShareButtonContainer from '../../share/ShareButton/ShareButtonContainer'
import VersoWallet from './wallet/VersoWalletContainer'
import Favorite from './favorite/FavoriteContainer'
import CancelThisLinkContainer from './booking/cancel-this-link/CancelThisLinkContainer'
import BookThisLinkContainer from './booking/book-this-link/BookThisLinkContainer'

const VersoControl = ({ booking, isFinished }) => (
  <div className="verso-control is-relative">
    <ul className="py8 px12 is-medium is-flex flex-0 flex-between items-center pc-theme-red">
      <li>
        <VersoWallet />
      </li>
      <li>
        <Favorite />
      </li>
      <li>
        <ShareButtonContainer />
      </li>
      <li className="is-relative">
        {booking && <CancelThisLinkContainer
          booking={booking}
          isFinished={isFinished}
                    />}
        {!booking && <BookThisLinkContainer isFinished={isFinished} />}
      </li>
    </ul>
    <Finishable finished={isFinished} />
  </div>
)

VersoControl.defaultProps = {
  booking: null,
  isFinished: false,
}

VersoControl.propTypes = {
  booking: PropTypes.shape(),
  isFinished: PropTypes.bool,
}

export default VersoControl
