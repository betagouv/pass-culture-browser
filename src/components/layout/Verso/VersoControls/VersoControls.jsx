import React from 'react'
import PropTypes from 'prop-types'

import BookingActionContainer from './booking/BookingAction/BookingActionContainer'
import CancellingActionContainer from './booking/CancellingAction/CancellingActionContainer'
import FavoriteContainer from './Favorite/FavoriteContainer'
import WalletContainer from './Wallet/WalletContainer'
import FinishableContainer from '../../Finishable/FinishableContainer'
import ShareButtonContainer from '../../Share/ShareButton/ShareButtonContainer'

const VersoControls = ({ isBooked, trackInFavorite }) => (
  <FinishableContainer>
    <ul className="verso-controls">
      <li>
        <WalletContainer />
      </li>
      <li>
        <FavoriteContainer trackInFavorite={trackInFavorite} />
      </li>
      <li>
        <ShareButtonContainer />
      </li>
      <li>
        {isBooked ? <CancellingActionContainer /> : <BookingActionContainer />}
      </li>
    </ul>
  </FinishableContainer>
)

VersoControls.defaultProps = {
  isBooked: false,
}

VersoControls.propTypes = {
  isBooked: PropTypes.bool,
  trackInFavorite: PropTypes.func.isRequired,
}

export default VersoControls
