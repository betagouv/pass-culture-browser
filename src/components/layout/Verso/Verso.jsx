import PropTypes from 'prop-types'
import React from 'react'

import VersoContentOfferContainer from './VersoContent/VersoContentOffer/VersoContentOfferContainer'
import VersoControlsContainer from './VersoControls/VersoControlsContainer'
import VersoHeaderContainer from './VersoHeader/VersoHeaderContainer'

const Verso = ({
  areDetailsVisible,
  extraClassName,
  offerName,
  offerType,
  offerVenueNameOrPublicName,
}) => (
  <div className={`verso is-overlay ${extraClassName} ${areDetailsVisible ? 'flipped' : ''}`}>
    <div className="verso-wrapper">
      <VersoHeaderContainer
        subtitle={offerVenueNameOrPublicName}
        title={offerName}
        type={offerType}
      />
      <VersoControlsContainer />
      <div className="mosaic-background verso-content">
        <VersoContentOfferContainer />
      </div>
    </div>
  </div>
)

Verso.defaultProps = {
  extraClassName: '',
  offerName: null,
  offerType: null,
  offerVenueNameOrPublicName: null,
}

Verso.propTypes = {
  areDetailsVisible: PropTypes.bool.isRequired,
  extraClassName: PropTypes.string,
  offerName: PropTypes.string,
  offerType: PropTypes.string,
  offerVenueNameOrPublicName: PropTypes.string,
}

export default Verso
