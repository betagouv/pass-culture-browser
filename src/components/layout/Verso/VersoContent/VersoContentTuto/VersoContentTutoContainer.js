import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import { compose } from 'redux'

import VersoContentTuto from './VersoContentTuto'
import { selectThumbUrlByRouterMatch } from '../../../../../selectors/data/thumbUrlSelector'

const VERSO_OVH_THUMB_SUFFIX = '_1'

export const mapStateToProps = (state, ownProps) => {
  const { match } = ownProps
  const thumbUrl = selectThumbUrlByRouterMatch(state, match)
  return {
    imageURL: `${thumbUrl}${VERSO_OVH_THUMB_SUFFIX}`,
  }
}

export default compose(
  withRouter,
  connect(mapStateToProps)
)(VersoContentTuto)
