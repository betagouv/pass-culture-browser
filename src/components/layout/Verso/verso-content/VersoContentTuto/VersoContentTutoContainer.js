import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import { compose } from 'redux'

import VersoContentTuto from './VersoContentTuto'
import selectThumbUrlByRouterMatch from '../../../../../selectors/selectThumbUrlByRouterMatch'

const mapStateToProps = (state, ownProps) => {
  const { match } = ownProps
  const thumbUrl = selectThumbUrlByRouterMatch(state, match)
  return {
    imageURL: thumbUrl,
  }
}

export default compose(
  withRouter,
  connect(mapStateToProps)
)(VersoContentTuto)
