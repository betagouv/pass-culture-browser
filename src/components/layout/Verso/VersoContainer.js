import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import { compose } from 'redux'

import Verso from './Verso'
import getHeaderColor from '../../../utils/colors'
import selectMediationByRouterMatch from '../../../selectors/selectMediationByRouterMatch'
import selectOfferByRouterMatch from '../../../selectors/selectOfferByRouterMatch'
import { ROOT_PATH } from '../../../utils/config'

const backgroundImage = `url('${ROOT_PATH}/mosaic-k.png')`

export const checkIsTuto = mediation => {
  const { tutoIndex } = mediation || {}
  const result = Boolean(typeof tutoIndex === 'number')
  return result
}

export const getContentInlineStyle = (isTuto, backgroundColor) => {
  let result = { backgroundImage }
  if (isTuto && backgroundColor) result = { ...result, backgroundColor }
  return result
}

const mapStateToProps = (state, ownProps) => {
  const { match } = ownProps
  const offer = selectOfferByRouterMatch(state, match) || {}
  const mediation = selectMediationByRouterMatch(state, match) || {}
  const firstThumbDominantColor = offer.firstThumbDominantColor || mediation.firstThumbDominantColor
  const backgroundColor = getHeaderColor(firstThumbDominantColor)
  const isTuto = checkIsTuto(mediation)
  const contentInlineStyle = getContentInlineStyle(isTuto, backgroundColor)

  const { name: offerName, venue } = offer
  const { name: venueName, publicName: venuePublicName } = venue || {}
  const offerVenueNameOrPublicName = venuePublicName || venueName

  return {
    backgroundColor,
    contentInlineStyle,
    isTuto,
    offerName,
    offerVenueNameOrPublicName,
  }
}

export default compose(
  withRouter,
  connect(mapStateToProps)
)(Verso)
