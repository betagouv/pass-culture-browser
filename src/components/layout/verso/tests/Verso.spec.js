import React from 'react'
import { shallow } from 'enzyme'

import Verso from '../Verso'
import VersoContentOfferContainer from '../verso-content/verso-content-offer/VersoContentOfferContainer'
import VersoContentTuto from '../verso-content/VersoContentTuto'
import VersoControls from '../verso-controls/VersoControls'

const backgroundColor = '#ACE539'
const props = {
  areDetailsVisible: true,
  backgroundColor,
  contentInlineStyle: { backgroundColor, backgroundImage: 'any/image' },
  extraClassName: 'extra-classname',
  forceDetailsVisible: false,
  mediationId: 'AAA',
  offerName: 'Offer title',
  offerVenueNameOrPublicName: 'Offer subtitle',
  recommendation: {
    mediation: {},
    thumbUrl: 'https://example.net/tuto/image.png',
  },
}

describe('src | components | verso | Verso', () => {
  it('should match snapshot', () => {
    // when
    const wrapper = shallow(<Verso {...props} />)

    // then
    expect(wrapper).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })

  it('should show offer view when is not tuto', () => {
    // when
    const wrapper = shallow(<Verso {...props} />)
    const infos = wrapper.find(VersoContentOfferContainer)
    const tuto = wrapper.find(VersoContentTuto)
    const controls = wrapper.find(VersoControls)

    // then
    expect(tuto).toHaveLength(0)
    expect(infos).toHaveLength(1)
    expect(controls).toHaveLength(1)
  })

  it('should show tuto view when is tuto', () => {
    // given
    props.recommendation.mediation.tutoIndex = 3

    // when
    const wrapper = shallow(<Verso {...props} />)
    const infos = wrapper.find(VersoContentOfferContainer)
    const tuto = wrapper.find(VersoContentTuto)
    const controls = wrapper.find(VersoControls)

    // then
    expect(tuto).toHaveLength(1)
    expect(tuto.props()).toHaveProperty('imageURL', 'https://example.net/tuto/image.png')
    expect(infos).toHaveLength(0)
    expect(controls).toHaveLength(0)
  })
})
