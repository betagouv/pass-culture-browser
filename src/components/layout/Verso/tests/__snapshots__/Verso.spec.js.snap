// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | verso | Verso should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Verso
    areDetailsVisible={true}
    backgroundColor="#ACE539"
    contentInlineStyle={
      Object {
        "backgroundColor": "#ACE539",
        "backgroundImage": "any/image",
      }
    }
    extraClassName="extra-classname"
    forceDetailsVisible={false}
    imageURL=""
    isTuto={false}
    mediationId="AAA"
    offerName="Offer title"
    offerVenueNameOrPublicName="Offer subtitle"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="verso-wrapper is-black-text scroll-y flex-rows is-relative text-left"
        >
          <VersoHeader
            backgroundColor="#ACE539"
            subtitle="Offer subtitle"
            title="Offer title"
          />
          <withRouter(Connect(VersoControl)) />
          <div
            className="verso-content"
            style={
              Object {
                "backgroundColor": "#ACE539",
                "backgroundImage": "any/image",
              }
            }
          >
            <withRouter(Connect(VersoContentOffer)) />
          </div>
        </div>,
        <withRouter(Connect(Footer))
          borderTop={true}
          colored={true}
          id="verso-footer"
        />,
      ],
      "className": "verso is-overlay extra-classname flipped",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <VersoHeader
              backgroundColor="#ACE539"
              subtitle="Offer subtitle"
              title="Offer title"
            />,
            <withRouter(Connect(VersoControl)) />,
            <div
              className="verso-content"
              style={
                Object {
                  "backgroundColor": "#ACE539",
                  "backgroundImage": "any/image",
                }
              }
            >
              <withRouter(Connect(VersoContentOffer)) />
            </div>,
          ],
          "className": "verso-wrapper is-black-text scroll-y flex-rows is-relative text-left",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "backgroundColor": "#ACE539",
              "subtitle": "Offer subtitle",
              "title": "Offer title",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <withRouter(Connect(VersoContentOffer)) />,
                false,
              ],
              "className": "verso-content",
              "style": Object {
                "backgroundColor": "#ACE539",
                "backgroundImage": "any/image",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              false,
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "borderTop": true,
          "colored": true,
          "id": "verso-footer",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="verso-wrapper is-black-text scroll-y flex-rows is-relative text-left"
          >
            <VersoHeader
              backgroundColor="#ACE539"
              subtitle="Offer subtitle"
              title="Offer title"
            />
            <withRouter(Connect(VersoControl)) />
            <div
              className="verso-content"
              style={
                Object {
                  "backgroundColor": "#ACE539",
                  "backgroundImage": "any/image",
                }
              }
            >
              <withRouter(Connect(VersoContentOffer)) />
            </div>
          </div>,
          <withRouter(Connect(Footer))
            borderTop={true}
            colored={true}
            id="verso-footer"
          />,
        ],
        "className": "verso is-overlay extra-classname flipped",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <VersoHeader
                backgroundColor="#ACE539"
                subtitle="Offer subtitle"
                title="Offer title"
              />,
              <withRouter(Connect(VersoControl)) />,
              <div
                className="verso-content"
                style={
                  Object {
                    "backgroundColor": "#ACE539",
                    "backgroundImage": "any/image",
                  }
                }
              >
                <withRouter(Connect(VersoContentOffer)) />
              </div>,
            ],
            "className": "verso-wrapper is-black-text scroll-y flex-rows is-relative text-left",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "backgroundColor": "#ACE539",
                "subtitle": "Offer subtitle",
                "title": "Offer title",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <withRouter(Connect(VersoContentOffer)) />,
                  false,
                ],
                "className": "verso-content",
                "style": Object {
                  "backgroundColor": "#ACE539",
                  "backgroundImage": "any/image",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                false,
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "borderTop": true,
            "colored": true,
            "id": "verso-footer",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
